gensim/models/keyedvectors.py

```diff
@@ -154,6 +154,84 @@ def get_vector(self, entity):
         else:
             raise KeyError("'%s' not in vocabulary" % entity)
 
+    def add_entity(self, entity, weights, replace=False):
+        """Add entity vector in a manual way.
+        If `entity` is already in the vocabulary, old vector is keeped unless `replace` flag is True.
+
+        Parameters
+        ----------
+        entity : str
+            Entity specified by string tag.
+        weights : np.array
+            1D numpy array with shape (`vector_size`,)
+        replace: bool, optional
+            Boolean flag indicating whether to replace old vector if entity is already in the vocabulary.
+            Default, False, means that old vector is keeped.
+        """
+        self.add_entities([entity], weights.reshape(1, -1), replace=replace)
+
+    def add_entities(self, entities, weights, replace=False):
+        """Add entities and theirs vectors in a manual way.
+        If some entity is already in the vocabulary, old vector is keeped unless `replace` flag is True.
+
+        Parameters
+        ----------
+        entities : list of str
+            Entities specified by string tags.
+        weights: {list of numpy.ndarray, numpy.ndarray}
+            List of 1D np.array vectors or 2D np.array of vectors.
+        replace: bool, optional
+            Flag indicating whether to replace vectors for entities which are already in the vocabulary,
+            if True - replace vectors, otherwise - keep old vectors.
+        """
+        if isinstance(weights, list):
+            weights = np.array(weights)
+
+        in_vocab_mask = np.zeros(len(entities), dtype=np.bool)
+        in_vocab_idxs = []
+        out_vocab_entities = []
```

Comment:
This method might be simpler without separate `in_vocab_idxs` and `out_vocab_entities` – just driving those ops from the mask, using options like `where()` or `nonzero()`.